nameko/testing/pytest.py

```diff
@@ -125,59 +121,90 @@ def rabbit_config(request, rabbit_manager):
     }
 
     yield conf
+    rabbit_manager.delete_vhost(vhost)
 
-    pools.reset()  # close connections in pools
-
-    def retry(fn):
-        """ Barebones retry decorator
-        """
-        def wrapper():
-            max_retries = 3
-            delay = 1
-            exceptions = RuntimeError
-
-            counter = itertools.count()
-            while True:
-                try:
-                    return fn()
-                except exceptions:
-                    if next(counter) == max_retries:
-                        raise
-                    time.sleep(delay)
-        return wrapper
-
-    @retry
-    def check_connections():
-        """ Raise a runtime error if the test leaves any connections open.
-
-        Allow a few retries because the rabbit api is eventually consistent.
-        """
-        connections = get_rabbit_connections(conf['vhost'], rabbit_manager)
-        open_connections = [
-            conn for conn in connections if conn['state'] != "closed"
-        ]
-        if open_connections:
-            count = len(open_connections)
-            names = ", ".join(conn['name'] for conn in open_connections)
-            raise RuntimeError(
-                "{} rabbit connection(s) left open: {}".format(count, names))
-    try:
-        check_connections()
-    finally:
-        rabbit_manager.delete_vhost(vhost)
 
+@pytest.yield_fixture(autouse=True)
+def fast_teardown(request):
+    """
+    This fixture fixes the order of the `container_factory`, `runner_factory`
+    and `rabbit_config` fixtures to get the fastest possible teardown of tests
+    that use them.
 
-@pytest.fixture
-def ensure_cleanup_order(request):
-    """ Ensure ``rabbit_config`` is invoked early if it's used by any fixture
-    in ``request``.
+    Without this fixture, the teardown order depends on the fixture resolution
+    defined by the test, for example::
+
+    def test_foo(container_factory, rabbit_config):
+        pass  # rabbit_config tears down first
+
+    def test_bar(rabbit_config, container_factory):
+        pass  # container_factory tears down first
+
+    This fixture ensures the teardown order is:
+
```

Comment:
I think it's because of the autouse and the fact that the "reorder fixtures" are all in the same plugin module. I don't think it's an official behaviour that's guaranteed to work in future versions but it does for now.

It _might_ be possible to guarantee it with some deeper pytest magic (I'm following https://github.com/pytest-dev/pytest/issues/1216), but I think this is OK for now given that we pin pytest.